src/dss_cell_group.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <cell_group.hpp>
+#include <dss_cell.hpp>
+#include <util/span.hpp>
+#include <util/unique_any.hpp>
+
+
+#include <iostream>
+namespace nest {
+namespace mc {
+
+/// Cell_group to collect spike sources
+class dss_cell_group : public cell_group {
+public:
+    using source_id_type = cell_member_type;
+
+    dss_cell_group(cell_gid_type first_gid, const std::vector<util::unique_any>& cell_descriptions):
+        gid_base_(first_gid)
+    {
+        using util::make_span;
+
+        for (cell_gid_type i: make_span(0, cell_descriptions.size())) {
+            // Copy all the rss_cells
+            cells_.push_back(dss_cell(
+                util::any_cast<dss_cell::dss_cell_description>(cell_descriptions[i])
+            ));
+
+            // create a lid to gid map
+            spike_sources_.push_back({gid_base_+i, 0});
+        }
+    }
+
+    virtual ~dss_cell_group() = default;
+
+    cell_kind get_cell_kind() const override {
+        return cell_kind::data_spike_source;
+    }
+
+    void reset() override {
+        for (auto cell: cells_) {
+            cell.reset();
+        }
+    }
+
+    void set_binning_policy(binning_kind policy, time_type bin_interval) override
+    {} // Nothing to do?
+
+    void advance(time_type tfinal, time_type dt) override {
+        // TODO: Move source information to dss_cell implementation
```

Comment:
Do we need the `dss_cell` at all? We are unlikely to have different "back ends" for this, so it could be simpler to put all of the logic in the `cell_group`.